lib/TemperatureSensor.js

```diff
@@ -0,0 +1,277 @@
+            );
+        }
+        return tempMins;
+    }
+
+    function getAmbientTemperatureMin(){
+        return Number(phidget.data.board.AmbientTemperatureMin);
+    }
+
+    function getAmbientTemperatureMax(){
+        return Number(phidget.data.board.AmbientTemperatureMax);
+    }
+
+    function getPotential(){
+        var pot=[];
+        for(var i in phidget.data.Potential){
+            pot.push(
+                Number(phidget.data.Potential[i])
+            );
+        }
+        return pot;
+    }
+
+    function getPotentialMin(){
+        var potMin;
+        potMin = Number(phidget.data.board.PotentialMin);
+        return potMin;
+    }
+
+    function getPotentialMax(){
+        var potMax;
+        potMax = Number(phidget.data.board.PotentialMax);
+        return potMax;
+    }
+
+    function getThermocoupleType(){
+        var type=[];
+        for(var i in phidget.data.ThermocoupleType){
+            type.push(
+                Number(phidget.data.ThermocoupleType[i])
+            );
+        }
+        return type;
+    }
+
+    function setThermocoupleType(changes){
+        // phidget.set(
+        //     {
+        //         type:'ThermocoupleType',
+        //         key:i.toString(),
+        //         value:board.Outputs[i].toString()
+        //     }
+        // );
+    }
+
+    phidget.params={
+        type:this.type
+    }
+
+    var readyHandler=false;
+    function ready(handler){
+        if(!handler){
+            return;
+        }
+
+        if(typeof handler!='function'){
+            return;
+        }
+
+        readyHandler=handler;
+    }
+
+    function observeTemperature(callback){
+        //console.log(temp);
+        if(typeof callback != 'function'){
+            throw('Requires a callback function as paramater');
+        }
+
+        Object.observe(
+            temps.temperature,
+            callback
+        );
+    }
+
+    function observeAmbientTemperature(callback){
+        if(typeof callback != 'function'){
+            throw('Requires a callback function as paramater');
+        }
+
+        Object.observe(
+            temps,
+            callback
+        );
+    }
+
+    phidget.on(
+        'log',
+        function(data){
+            //log it?
+            //console.log('log ',data);
+        }
+    );
+
+    phidget.on(
+        'error',
+        function(data){
+            //throw it?
+            //console.log('error ',data);
+        }
+    );
+
+    phidget.on(
+        'changed',
+        update
+    );
+
+    function update(data){
+        try{
+            for(var i in phidget.data.Temperature){
+                temps.temperature[i]=Number(phidget.data.Temperature[i]);
+            }
+            for(var i in phidget.data.board.AmbientTemperature){
+                temps.ambientTemp=Number(phidget.data.board.AmbientTemperature);
+            }
+        }catch(err){
+            //probably doesn't exist yet
+        }
+    }
```

Comment:
This should be with the bulk of the code instead of down in the guts of the boilerplate stuff.